### ISSUE TITLE
Proxy hosts have to be updated on container start/restart to work properly

### ISSUE THREAD

**Issue by NateTheSage** [NginxProxyManager/nginx-proxy-manager#2171]
**Checklist**
- Have you pulled and found the error with `jc21/nginx-proxy-manager:latest` docker image?
  - Yes
- Are you sure you're not using someone else's docker image?
  - Yes
- Have you searched for similar issues (both open and closed)?
  - Yes

**Describe the bug**
I can't recall when this started happening, but apparently whenever I stop and start the container again, regardless of what happened to cause the stop or start (i.e. host reboot), all proxy hosts saved do not work, and every single page returned by the proxy hosts is the congratulations page if on HTTP. HTTPS will not connect and on Firefox will say "Secure Connection Failed". I have to go into the admin portal and resave every single proxy host in order to get them all working again. I've had friends also report this to me with some other services that I proxy after my VM or the container has rebooted.

**Nginx Proxy Manager Version**
v2.9.18

**To Reproduce**
1. Create a proxy host to an existing service
2. Verify proxy host works from a connection external to the network
3. `docker-compose down` & `docker-compose up -d`
4. Proxy host no longer works on the external connection and displays the "Congratulations!" page over HTTP. HTTPS does not complete a connection and will say something to the effect of "Secure Connection Failed" in the browser.
5. Log back into the admin UI, pick the proxy host that just was created, hit "Edit".
6. Hit "Save".
7. Proxy host now works again from the external connection.

**Expected behavior**
Proxy hosts should work after container reboot.

**Screenshots**
The external device I am using is a Moto x4 handset's LTE connection as I've got no other way to view this page from the outside.

Here's an example, showing my XBackBone login when the proxy is working.
![Screenshot_20220725-095139_crDroid_Home](https://user-images.githubusercontent.com/42617513/180832324-a9a1b3f4-825d-4c26-b7dd-be8d098755b9.png)

When the proxy is not, this is what I see.
![Screenshot_20220725-104829_crDroid_Home](https://user-images.githubusercontent.com/42617513/180832448-37194157-5c76-45b3-a975-88e59ea6b7ba.png)

(Further note, there is a rewrite in place that will rewrite HTTP requests to HTTPS, I don't have an automatic HTTPS rewriter addon for my Android Firefox.)

**Operating System**
Debian 11 VM on an ESXi 7.0 Dell R720.

**Additional context**
Docker version: `Docker version 20.10.5+dfsg1, build 55c4c88`

docker-compose:
```
version: '3'
services:
  app:
    image: 'jc21/nginx-proxy-manager:latest'
    restart: always
    environment:
      DISABLE_IPV6: 'true'
      DB_MYSQL_HOST: "db"
      DB_MYSQL_PORT: 3306
      DB_MYSQL_USER: "npm"
      DB_MYSQL_PASSWORD: "<SNIP>"
      DB_MYSQL_NAME: "npm"
    restart: unless-stopped
    ports:
      - '80:80'
      - '81:81'
      - '443:443'
    volumes:
      - ./data:/srv/nginxproxymanager/data
      - ./letsencrypt:/etc/letsencrypt
    depends_on:
      - db

  db:
    image: 'jc21/mariadb-aria:latest'
    restart: always
    environment:
      MYSQL_ROOT_PASSWORD: '<SNIP>'
      MYSQL_DATABASE: 'npm'
      MYSQL_USER: 'npm'
      MYSQL_PASSWORD: '<SNIP>'
    volumes:
      - ./data/mysql:/var/lib/mysql
```
nginxproxymanager systemd unit file:
```
[Unit]
Description=nginxproxymanager
Requires=docker.service
After=docker.service

[Service]
Restart=always
User=root
Group=docker
WorkingDirectory=/srv/nginxproxymanager
# Shutdown container (if running) when unit is started
ExecStartPre=/usr/bin/docker-compose -f docker-compose.yml down
# Start container when unit is started
ExecStart=/usr/bin/docker-compose -f docker-compose.yml up
# Stop container when unit is stopped
ExecStop=/usr/bin/docker-compose -f docker-compose.yml down

[Install]
WantedBy=multi-user.target
```

Both my `docker logs` for the `app` and the `db` containers have no errors in them, but should any other logging or the like be needed I'd be happy to provide it.

**Comment by tralce** [comment]
I seem to also be having this issue. However, instead of showing the Congratulations page, http://host still points at the non-ssl version of the host (even though Force SSL is on). My "Default page" is set to "Custom page" and I've just put some very basic HTML in there. https://host fails as Nate documented. 

If I `curl https://host`, curl reports: 
```curl: (60) SSL certificate problem: self signed certificate
More details here: https://curl.se/docs/sslcerts.html
curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.
```

Excerpt from my compose fileL
```  nginx-proxy-manager:                                                                                                                                                                                                      
    image: 'jc21/nginx-proxy-manager:latest'
    container_name: nginx-proxy-manager
    ports:
      - '80:80'
      - '81:81'
      - '443:443'
    volumes:
      - /home/tralce/volumes/nginx-proxy-manager/data:/data
      - /home/tralce/volumes/nginx-proxy-manager/letsencrypt:/etc/letsencrypt
    restart: unless-stopped```

**Comment by RobertYoung** [comment]
I am also seeing this issue. The conf file for the proxy host doesn't have the ssl information that the database has defined. Saving the proxy host again via the UI re-renders the conf with the correct information

**Comment by NateTheSage** [comment]
> I am also seeing this issue. The conf file for the proxy host doesn't have the ssl information that the database has defined. Saving the proxy host again via the UI re-renders the conf with the correct information. It seems to happen whenever the certificates are renewed by letsencrypt

This is what I suspect as well, except for me it's not even persisting across a container kick. I've been working for some time to try and make a bandage for this so I don't have to go in and resave every single host if I need to kick either the container or the underlying host for updates. I've only got 3 right now, but this is keeping me from defining more hosts purely from the amount of clicking I'd have to do.

**Comment by NateTheSage** [comment]
Oh. Turns out I'm not as smart as I thought I was. I had set this up initially when I was just learning `docker-compose`, and realized now that I've learned a few things that my data volume:

`- ./data:/srv/nginxproxymanager/data`

Needed to look something more akin to this:

`- ./data:/data`

I had *totally* misunderstood the setup process described because I didn't fully understand how `docker-compose` worked.

Fixing that, I now see in my volume directory:

```
drwxr-xr-x 2 root root 4.0K Jul 26 16:25 .
drwxr-xr-x 9 root root 4.0K Jul 26 16:21 ..
-rw-r--r-- 1 root root 1.1K Jul 26 16:25 1.conf
-rw-r--r-- 1 root root 1.3K Jul 26 16:25 2.conf
-rw-r--r-- 1 root root 1.3K Jul 26 16:25 4.conf
```
The conf files for my 3 redirection hosts.

Turns out it was a PEBKAC on my part. 🥇
